File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashMap = {}
        if len(s) != len(t):
            return False
        for char in s:
            hashMap[char] = hashMap.get(char, 0) + 1
        for char2 in t:
            if char2 in hashMap:
                if hashMap.get(char2, 0) == 0:
                    return False
                else:
                    hashMap[char2] = hashMap.get(char2, 0) - 1
            else:
                return False
        return True

            

File: Data Structures & Algorithms/valid-sudoku/submission-1.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Map<Integer, Set<Character>> colMap = new HashMap<>();
        Map<List<Integer>, Set<Character>> subBoxMap = new HashMap<>();

        for (int row=0; row < board.length; row++) {
            Set<Character> rowSet = new HashSet<>();

            for (int col=0; col < board[row].length; col++){
                char c = board[row][col];

                if (c == '.') continue;

                if (c < '1' || c > '9') return false;

                if (!rowSet.add(c)) return false;

                if (!colMap.computeIfAbsent(col, k -> new HashSet<>()).add(c)) 
                    return false;
                
                if (!subBoxMap.computeIfAbsent(List.of(row/3, col/3), k -> new HashSet<>()).add(c)) 
                    return false;
            }
        }

        return true;
    }
}
